### ISSUE TITLE
Pipeline

### ISSUE THREAD

**Issue by qrb** [imjacobclark/Nodezilla#7]
You're gonna want some tests right?
What are the most crushing concerns you need to cover, do you want a hand with anything in particular?

**Comment by imjacobclark** [comment]
Yeah, my next concern is to write tests, been a little tied up to work on this recently but before I write tests I was looking to get the core functionality implemented, including:

* Basic charts
* Basic data represented correctly, there are a few bugs currently related to how the average data is calculated 
* A better structured UI, Angular, Backbone? 

Once those three things are implemented, I intend to write tests and cut a release before moving on to implementing load time testing and more test strategies too.

I'm also slightly concerned with the Socket.io implementation, a known issue I've been seeing is where Socket is falling back to XHR requests opposed to WebSocket for data transfer, I need to look into this.

Feel free to implement functionality or write tests, I'm happy to review pull requests and merge them in.
